src/Sarif/Baseline/V2/WhereComparer.cs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+using Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching;
+
+namespace Microsoft.CodeAnalysis.Sarif.Baseline
+{
+    public class WhereComparer : IComparer<ExtractedResult>
+    {
+        public static WhereComparer Instance = new WhereComparer();
+
+        public int Compare(ExtractedResult left, ExtractedResult right)
+        {
+            return CompareWhere(left, right);
+        }
+
+        public static int CompareWhere(ExtractedResult left, ExtractedResult right)
+        {
+            return CompareTo(left?.Result?.Locations, left?.OriginalRun, right?.Result?.Locations, right?.OriginalRun);
+        }
+
+        public static int CompareTo(IList<Location> left, Run leftRun, IList<Location> right, Run rightRun)
+        {
+            if (left == null && right == null) { return 0; }
+            if (left == null) { return -1; }
+            if (right == null) { return 1; }
+
+            int commonLength = Math.Min(left.Count, right.Count);
+            for (int i = 0; i < commonLength; ++i)
+            {
+                int cmp = CompareTo(left[i], leftRun, right[i], rightRun);
```

Comment:
>left [](start = 36, length = 4)

Are the Left and Right lists sorted? i.e. doesn't the comparison rely on the lists being sorted?

Reply:
Hmm. They aren't, and they should be. I don't know how to easily sort them, because an IComparer<Location> doesn't have the accompanying run to look up locations which just have an index. I think we need to figure out how to deal with that in the OM before we can really improve this.